client.js

```diff
@@ -46,14 +46,15 @@ function checkMessages() {
             setTimeout(checkMessages, 100)
             return
         }
-        var estimatedTime = (request.response.byteLength / 4) / 4.41 + 2
-        setTimeout(checkMessages, estimatedTime)
+//        var estimatedTime = (request.response.byteLength / 4) / 4.41 + 2
+//        setTimeout(checkMessages, estimatedTime)
 
         console.log('received ' + request.response.byteLength)
         var source = playbackCtx.createBufferSource()
         console.log('start playback')
         source.onended = function() {
             console.log('end playback')
+            setTimeout(checkMessages, 1000) // aici ar trebui setat TimeOut-ul corect
```

Comment:
Okay, I believe you. Dar nu mi-ai răspuns la întrebare? Dacă dă serverul timeout sau ai glitch la net se cheamă și .onended sau doar .onerror?